### PR TITLE
Fast browser tests

### PR DESCRIPTION
**Pull request by kripken**
This makes the speed of running simple tests almost 2x faster - you can really notice this when running the tests, there are no odd stalls for no reason. Overall bot test times improve 33% (since a significant amount of time is spent in compilation, which has not changed).

What this PR does is replace two http servers with one. We had one for the harness, and created one for each test. This PR uses a single persistent one, so we save creating the per-test one on every test. That saves process creation overhead etc.

This also changes the polling speed that the JS code in the harness uses. This has a smaller effect but is still useful.

Also fix `test_preload_caching` for chrome, we must avoid trying to cache very large files there. Not sure what that started to fail here more - perhaps timing related, e.g. maybe chrome needs time between tests for something to happen to the IndexedDB cache. I made sure the test checks a bunch of interesting but smaller sizes.

### CODE REVIEW

File: tests/browser_harness.html [hunk 1]
```diff
@@ -23,17 +23,24 @@
...
     function check() {
       var request = new XMLHttpRequest();
       request.open('GET', '/check');
       request.send(null);
       request.addEventListener("load", function() {
-        if (request.responseText != 'False') {
-          iframe.src = request.responseText;
+        if (request.responseText.startsWith(COMMAND_PREFIX)) {
+          var url = request.responseText.substr(COMMAND_PREFIX.length);
+          console.log('commanded to load ' + url);
```
Comment (juj) [inline tests/browser_harness.html]: Is this log intended to be permanent, or was this used temporarily for debugging? (good either way)

**Comment by juj** [comment]
lgtm! Browser tests are definitely the slowest to run, good to get speed to them.

File: tests/browser_harness.html (hunk 1 above)
Comment (kripken) [reply tests/browser_harness.html]: Hmm, yeah, kind of both really. I can remove later if it's annoying.

Comment (juj) [reply tests/browser_harness.html]: Looking at https://circleci.com/gh/emscripten-core/emscripten/91721?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link this does have the effect of over-verbosing the print, so CI says

```
ok
test_sdl_canvas_proxy (test_browser.browser) ... [browser launch: test.html ]
[queue command: http://localhost:8888/test.html /tmp/emscripten_test_browser_kzxUvb ]
[simple HTTP serving: /test.html ]
[simple HTTP serving: /test.js ]
[simple HTTP serving: /test.data ]
[simple HTTP serving: /test.wasm ]
[simple HTTP serving: /sdl_canvas_proxYour output is too large to display in the browser. Only the first 400000 characters are displayed.
Download the full output as a file.
```

perhaps good to drop the log it looks like.

Comment (kripken) [reply tests/browser_harness.html]: Thanks, yeah, that is annoying. Fix in #8062

**Pull request merged**